SlowCheetah.JDT/Processors/JdtMerge.cs

```diff
@@ -0,0 +1,113 @@
+
+        /// <inheritdoc/>
+        public override void Process(JObject source, JObject transform)
+        {
+            JToken mergeValue;
+            if (transform.TryGetValue(JsonUtilities.JdtSyntaxPrefix + this.Verb, out mergeValue))
+            {
+                this.Merge(source, mergeValue, true);
+            }
+
+            this.Successor.Process(source, transform);
+        }
+
+        private void Merge(JObject source, JToken mergeValue, bool allowArray)
+        {
+            switch (mergeValue.Type)
+            {
+                case JTokenType.Array:
+                    if (allowArray)
+                    {
+                        // If the value is an array, perform the replace for each object in the array
+                        foreach (JToken arrayValue in (JArray)mergeValue)
+                        {
+                            this.Merge(source, arrayValue, false);
+                        }
+                    }
+                    else
+                    {
+                        JsonUtilities.ThrowIfRoot(source, "Cannot replace root");
+                        source.Replace(mergeValue);
+                    }
+
+                    break;
+                case JTokenType.Object:
+                    this.MergeWithObject(source, (JObject)mergeValue);
+                    break;
+                default:
+                    JsonUtilities.ThrowIfRoot(source, "Cannot replace root");
+                    source.Replace(mergeValue);
+                    break;
+            }
+        }
+
+        private void MergeWithObject(JObject source, JObject mergeObject)
+        {
+            JToken pathToken, valueToken;
+            string pathFullAttribute = JsonUtilities.JdtSyntaxPrefix + PathAttribute;
+            bool hasPath = mergeObject.TryGetValue(pathFullAttribute, out pathToken);
+            string valueFullAttribute = JsonUtilities.JdtSyntaxPrefix + ValueAttribute;
+            bool hasValue = mergeObject.TryGetValue(valueFullAttribute, out valueToken);
+
+            if (!hasPath && !hasValue)
+            {
+                // If the merge object does not contain attributes,
+                // simply execute the transform with that object
+                ProcessTransform(source, mergeObject);
+            }
+            else if (hasPath && hasValue)
+            {
+                if (mergeObject.Properties().Where(p => !p.Name.Equals(pathFullAttribute) && !p.Name.Equals(valueFullAttribute)).Count() > 0)
+                {
+                    throw new JdtException("Merge only accepts path and value attributes");
+                }
+
+                if (pathToken.Type != JTokenType.String)
+                {
+                    throw new JdtException("Path attribute must be a string");
+                }
+
+                var tokensToMerge = JsonUtilities.GetTokensFromPath(source, pathToken.ToString());
+                foreach (JToken tokenToMerge in tokensToMerge.ToList())
```

Comment:
`IEnumerable` does not support altering the contents. I'll just have the utility method return the list